File: UCF HSPT Documents/2015/Solutions/fractal.c
/**************************
*
* In this problem we are given a number represented as the sum of two infinite fractions
* and are asked to determine its value. The key to solving this problem is to observe that
* because the denominators of the two fractions continue infinitely, they are actually equivalent
* to x. Because of this, we can rewrite the equation given to us in the following way:
*
*	x   =  a / x + b / x
*	x   = (a + b) / x
*	x^2 = (a + b)
*	x   =  sqrt(a + b)
*
* So, through a little algebra, we've simplified the equation given to us into one we can easily
* evaluate. Since the square root of (a + b) will always be a real number if (a + b) > 0, we only
* need to output "DNE" if the opposite is true.
*
**************************/


#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(){
	//Declare variables
	int times = 0, i = 0, a = 0, b = 0;
	
	//Read number of cases
	scanf("%d", &times);
	
	//Loop through cases
	for(i = 1; i <= times; i++){
		//Read a and b
		scanf("%d %d", &a, &b);
		
		//Perform DNE check, as described above
		if(a + b <= 0){
			//x is not a real number!
			printf("Fraction #%d: DNE\n", i);
		}else{
			//x must be the square root of (a + b), as described above
			printf("Fraction #%d: %0.2lf\n", i, sqrt(a + b));
		}
	}
	
	return 0;
}

File: UCF HSPT Team/src/y2015/fractal.java
import java.util.Scanner;

public class fractal {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int runs = scan.nextInt();

		for(int caseNum = 1; caseNum <= runs; caseNum++) {
			int a = scan.nextInt(), b = scan.nextInt();
			/**
			  * This whole problem simplifies down to the equation x = a/x + b/x after substituting in x for the denominators of the expression.
			  * Some algebra will rearrange the equation to x^2 = a + b or x = sqrt(a + b).
			  * Since we can't take the square root of a negative number, x doesn't exist if the sum of a and b is negative.
			  */
			if(a + b > 0)
				System.out.printf("Fraction #%d: %.2f\n", caseNum, Math.sqrt(a + b));
			else
				System.out.printf("Fraction #%d: DNE\n", caseNum);
		}
		scan.close();
	}
}
